src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2168,13 +2218,35 @@ public int GetLowerBound(int dimension)
 
             if (this.Rank != 1)
                 throw new PlatformNotSupportedException(SR.Rank_MultiDimNotSupported);
+#endif
             if (dimension != 0)
                 throw new IndexOutOfRangeException();
             return 0;
         }
 
         public int GetUpperBound(int dimension)
         {
+#if CORERT
```

Comment:
I think it would be worth it to have unique #define for it, so that we have easier time cleaning it up later once we unify.